File: src/go-backend/internal/models/category.go
package models

// URL params
type CategoryRequest struct {
	ServiceTag string
	CategoryID string
}

type ServiceCategoriesRequest struct {
	ServiceTag string
}

// Category Response
type Category struct {
	BackdropURL string `json:"backdropURL"`
	ID          string `json:"id"`
	Name        string `json:"name"`
	Overview    string `json:"overview"`
	PosterURL   string `json:"posterURL"`
	Shows       []Show `json:"shows"`
}

// Categories Response
type Categories struct {
	CategoryCount int        `json:"categoryCount"`
	Categories    []Category `json:"categories"`
}

File: src/frontend/src/lib/types/category.ts
import type { Show } from "./show";

export type CategoryRequest = {
    ServiceTag: string;
    CategoryID: string;
};

export type ServiceCategoriesRequest = {
    ServiceTag: string;
};

export type Category = {
    backdropURL: string;
    id: string;
    name: string;
    overview: string;
    posterURL: string;
    shows: Show[];
    serviceTag?: string;
};

export type Categories = {
    categoryCount: number;
    categories: Category[];
};
